source/tests/system/nirfmxspecan_driver_api_tests.cpp

```diff
@@ -445,6 +447,52 @@ TEST_F(NiRFmxSpecAnDriverApiTests, BuildSpurString_ReturnsSpurString)
   EXPECT_SUCCESS(session, spur_string_response);
 }
 
+TEST_F(NiRFmxSpecAnDriverApiTests, DefaultConfiguration_IQFetchData_RecordIsDeleted)
+{
+  constexpr auto INVALID_RECORD = -379451;
+  const auto session = init_session(stub(), PXI_5663);
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MeasurementTypes::MEASUREMENT_TYPES_IQ, true));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+
+  const auto fetch_response = EXPECT_SUCCESS(session, client::iq_fetch_data(stub(), session, "", 10.0, 0, 1000));
+  EXPECT_THAT(fetch_response.data(), Not(IsEmpty()));
+
+  EXPECT_RFMX_ERROR(
+      INVALID_RECORD,
+      "The requested record number is invalid. The desired record was already fetched and deleted.",
+      session,
+      client::iq_fetch_data(stub(), session, "", 10.0, 0, 1000));
+}
+
+TEST_F(NiRFmxSpecAnDriverApiTests, DefaultConfiguration_IQFetchDataFetchAllAvailable_DataIsFetched)
+{
+  constexpr auto INVALID_RECORD = -379451;
+  constexpr auto FETCH_ALL_AVAILABLE = -1;
+  constexpr auto EXPECTED_RECORD_COUNT = 50000;
+  const auto session = init_session(stub(), PXI_5663);
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MeasurementTypes::MEASUREMENT_TYPES_IQ, true));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+
+  const auto fetch_response = EXPECT_SUCCESS(session, client::iq_fetch_data(stub(), session, "", 10.0, 0, FETCH_ALL_AVAILABLE));
+
+  EXPECT_THAT(fetch_response.data(), Not(IsEmpty()));
+  EXPECT_EQ(EXPECTED_RECORD_COUNT, fetch_response.data().size());
```

Comment:
Question: Why is this 50000?

Reply:
That's just the amount of data you get in the default configuration. I think it's consistent. Really what I care about is that it's a reasonable number that's bigger than the arbitrarily truncated reads in the other tests (`1000`).